yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1154,6 +1154,133 @@ def pixelize_sph_kernel_slice(
+
+    x_min = bounds[0]
+    x_max = bounds[1]
+    y_min = bounds[2]
+    y_max = bounds[3]
+    z_min = bounds[4]
+    z_max = bounds[5]
+
+    dx = (x_max - x_min) / xsize
+    dy = (y_max - y_min) / ysize
+    dz = (z_max - z_min) / zsize
+    idx = 1.0/dx
+    idy = 1.0/dy
+    idz = 1.0/dz
+
+    kernel_func = get_kernel_func(kernel_name)
+
+    # define this to avoid using the use_normalization python object
+    # in the tight loop
+    cdef np.intp_t use_norm = int(use_normalization)
+
+    with nogil:
+        for j in range(0, posx.shape[0]):
+            if j % 50000 == 0:
+                with gil:
+                    if(pbar is not None):
+                        pbar.update(50000)
+                    PyErr_CheckSignals()
+
+            x0 = <np.int64_t> ( (posx[j] - hsml[j] - x_min) * idx)
+            x1 = <np.int64_t> ( (posx[j] + hsml[j] - x_min) * idx)
+            x0 = iclip(x0-1, 0, xsize)
+            x1 = iclip(x1+1, 0, xsize)
+
+            y0 = <np.int64_t> ( (posy[j] - hsml[j] - y_min) * idy)
+            y1 = <np.int64_t> ( (posy[j] + hsml[j] - y_min) * idy)
+            y0 = iclip(y0-1, 0, ysize)
+            y1 = iclip(y1+1, 0, ysize)
+
+            z0 = <np.int64_t> ( (posz[j] - hsml[j] - z_min) * idz)
+            z1 = <np.int64_t> ( (posz[j] + hsml[j] - z_min) * idz)
+            z0 = iclip(z0-1, 0, zsize)
+            z1 = iclip(z1+1, 0, zsize)
+
+            h_j3 = fmax(hsml[j]*hsml[j]*hsml[j], dx*dy*dz)
+            h_j = math.cbrt(h_j3)
+            h_j2 = h_j*h_j
+            ih_j = 1/h_j
+
+            w_j = pmass[j] / pdens[j] / hsml[j]**3
+            coeff = w_j * quantity_to_smooth[j]
+
+            # Now we know which voxels to deposit onto for this particle,
+            # so loop over them and add this particle's contribution
+            for xi in range(x0, x1):
+                x = (xi + 0.5) * dx + x_min
+
+                posx_diff = posx[j] - x
+                posx_diff = posx_diff * posx_diff
+                if posx_diff > 2 * h_j2:
+                    continue
+
+                for yi in range(y0, y1):
+                    y = (yi + 0.5) * dy + y_min
+
+                    posy_diff = posy[j] - y
+                    posy_diff = posy_diff * posy_diff
+                    if posy_diff > 2 * h_j2:
+                        continue
+
+                    for zi in range(z0, z1):
+                        z = (zi + 0.5) * dz + z_min
+
+                        posz_diff = posz[j] - z
+                        posz_diff = posz_diff * posz_diff
+                        if posz_diff > 2 * h_j2:
+                            continue
+
+                        # yt's kernel functions use a different convention than
+                        # the SPLASH paper (following Gadget-2), and qxy is
+                        # twice the value of q used in the SPLASH paper
+                        qxy = 2.0 * math.sqrt(posx_diff + posy_diff + posz_diff)
```

Comment:
I think we should just call this `q` here and in the other SPH pixelizers. It's called `q` in the paper, so it makes it a little easier to compare with the paper. In addition, here z is included as well, so `qxy` is a bit of a misnomer.

I'd also add in the comment above to see equation 4 in the SPLASH paper.

Reply:
It also should be *one half* the value in the SPLASH paper, and the factor of 2.0 should be deleted. See the patch I sent you over slack.